jcache/remote/src/main/java/org/infinispan/jcache/remote/RemoteCacheWithStats.java

```diff
@@ -12,16 +14,25 @@ public RemoteCacheWithStats(RemoteCache<K, V> delegate, LocalStatistics stats) {
       this.stats = stats;
    }
 
-   @Override
-   public V get(Object key) {
-      V v = delegate.get(key);
+   private final Function<V, V> getFunction = v -> {
       if (v == null) {
          stats.incrementCacheMisses();
       } else {
          stats.incrementCacheHits();
       }
       stats.incrementCacheGets();
       return v;
+   };
+
+   @Override
+   public CompletableFuture<V> getAsync(K key) {
```

Comment:
are these methods needed? They don't seem to be invoked in `JCache`

Reply:
if they are, the `putAsyn()` isn't updating the stats.

Reply:
No I thought I reverted all the changes let me do that.

Reply:
Also, it is missing the following methods in this class:
```
@Override
   public boolean replace(K key, V oldValue, V newValue) {
      boolean replaced = delegate.replace(key, oldValue, newValue);
      if (replaced) {
         stats.incrementCachePuts();
      }
      return replaced;
   }
```
and
```
@Override
   public boolean remove(Object key, Object oldValue) {
      boolean removed = delegate.remove(key, oldValue);
      if (removed) {
         stats.incrementCacheRemovals();
      }
      return removed;
   }
```
It fixes the CacheMBStatisticsBeanTest

Reply:
The same methods are missing from `RemoteCacheWithSyncListeners`.

Reply:
Thanks, yeah removing the old value wrapper is causing that. The RemoteCacheImpl properly implements these methods out of the box now so that class wasn't needed any longer. But these were relying on it to translate the call to another method.